connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -562,8 +709,13 @@ ConfigInfos validateConnectorConfig(
             configKeys.putAll(configDef.configKeys());
             allGroups.addAll(configDef.groups());
             configValues.addAll(config.configValues());
-            ConfigInfos configInfos =  generateResult(connType, configKeys, configValues, new ArrayList<>(allGroups));
 
+            // do custom converter-specific validation
+            ConfigInfos headerConverterConfigInfos = validateHeaderConverterConfig(connectorProps, validatedConnectorConfig.get(HEADER_CONVERTER_CLASS_CONFIG));
+            ConfigInfos keyConverterConfigInfos = validateKeyConverterConfig(connectorProps, validatedConnectorConfig.get(KEY_CONVERTER_CLASS_CONFIG));
+            ConfigInfos valueConverterConfigInfos = validateValueConverterConfig(connectorProps, validatedConnectorConfig.get(VALUE_CONVERTER_CLASS_CONFIG));
```

Comment:
nit: these constants are provided here, and also hard-coded inside these functions

Reply:
Yeah, the alternative was to pass in the entire `validatedConnectorConfig` and let the various `validateXxxConverterConfig` methods pull out the relevant `ConfigValue` field. But this seemed strange considering those methods only require a single `ConfigValue` object.